### PR TITLE
Fix: create delete list integration tests

### PR DESCRIPTION
**Pull request by Tulsishah**
This will be pull request template
### Description
Fix:
- Separate Create and Delete flows
- CreateObjectTest - Updating name
- DeleteObjectTest - Updating name
- List Test - Updating condition from && to || (as any condition fails, it should fail)
- Updating const variable name

### Testing details
1. Manual - all the integration tests are passing
2. Unit tests - N/A
3. Integration tests - All the integration tests are passing in  [integration tests](https://github.com/GoogleCloudPlatform/gcsfuse/tree/master/tools/integration_tests) dir

### CODE REVIEW

File: tools/integration_tests/readonly/create_object_test.go [hunk 1]
```diff
@@ -19,80 +19,60 @@ import (
...
-	ensureFileSystemLockedForDirCreation(dirPath, t)
-}
-
-func ensureFileSystemLockedForDeletion(objPath string, t *testing.T) {
-	err := os.RemoveAll(objPath)
 
 	// It will throw an error read-only file system or permission denied.
-	if err == nil {
-		t.Errorf("Objects are deleted in read-only file system.")
+	if !strings.Contains(err.Error(), "read-only file system") && !strings.Contains(err.Error(), "permission denied") {
```
Comment (vadlakondaswetha) [inline tools/integration_tests/readonly/create_object_test.go:63]: do you want to extract this into seperate function in readonly_test.go file

File: tools/integration_tests/readonly/list_objects_test.go [hunk 2]
```diff
@@ -39,11 +39,11 @@ func TestListObjectsInBucket(t *testing.T) {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != DirectoryNameInTestBucket && obj[0].IsDir() != true {
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
```
Comment (vadlakondaswetha) [inline tools/integration_tests/readonly/list_objects_test.go:42]: please add a comment here explaining what you area validating

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: tools/integration_tests/readonly/delete_object_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+func checkIfObjDeletionFailed(objPath string, t *testing.T) {
+	err := os.RemoveAll(objPath)
+
+	if err == nil {
+		t.Errorf("Objects are deleted in read-only file system.")
+	}
+
+	// It will throw an error read-only file system or permission denied.
+	if !strings.Contains(err.Error(), "read-only file system") && !strings.Contains(err.Error(), "permission denied") {
```
Comment (sethiay) [inline tools/integration_tests/readonly/delete_object_test.go:35]: We are checking these errors in the same way at 3 places. Can we create a common function to check for these errors ?

File: tools/integration_tests/readonly/delete_object_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	checkIfObjDeletionFailed(objPath, t)
+}
+
+func TestDeleteSubDirectory(t *testing.T) {
+	objPath := path.Join(setup.MntDir(), DirectoryNameInTestBucket, SubDirectoryNameInTestBucket)
+
+	checkIfObjDeletionFailed(objPath, t)
+}
+
+func TestDeleteFileInSubDirectory(t *testing.T) {
```
Comment (sethiay) [inline tools/integration_tests/readonly/delete_object_test.go:58]: The name of test case is delete file in sub directory. However, the test case is deleting file in directory.

File: tools/integration_tests/readonly/create_object_test.go (hunk 1 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/create_object_test.go:60]: fixed

File: tools/integration_tests/readonly/list_objects_test.go (hunk 2 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go:45]: fixed

File: tools/integration_tests/readonly/create_object_test.go [hunk 5]
```diff
@@ -19,80 +19,60 @@ import (
...
 	file, err := os.OpenFile(filePath, os.O_CREATE, setup.FilePermission_0600)
 
-	// It will throw an error read-only file system or permission denied.
 	if err == nil {
 		t.Errorf("File is created in read-only file system.")
 	}
 
+	// It will throw an error read-only file system or permission denied.
+	if !strings.Contains(err.Error(), "read-only file system") && !strings.Contains(err.Error(), "permission denied") {
+		t.Errorf("Throwing incorrect error.")
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/readonly/create_object_test.go]: fixed

File: tools/integration_tests/readonly/list_objects_test.go [hunk 6]
```diff
@@ -64,11 +64,11 @@ func TestListObjectsInBucketSubDirectory(t *testing.T) {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != FileInSubDirectoryNameInTestBucket && obj[0].IsDir() != false {
+	if obj[0].Name() != FileNameInDirectoryTestBucket || obj[0].IsDir() != false {
 		t.Errorf("Listed object is incorrect.")
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: fixed

File: tools/integration_tests/readonly/delete_object_test.go (hunk 4 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/delete_object_test.go]: fixed

File: tools/integration_tests/readonly/delete_object_test.go (hunk 3 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/delete_object_test.go]: fixed

File: tools/integration_tests/readonly/list_objects_test.go [hunk 7]
```diff
@@ -60,15 +65,20 @@ func TestListObjectsInBucketSubDirectory(t *testing.T) {
...
+	if len(obj) != NumberOfObjectsInDirectoryTestBucket {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != FileInSubDirectoryNameInTestBucket && obj[0].IsDir() != false {
-		t.Errorf("Listed object is incorrect.")
+	// Comparing first object name and type.
+	// Name - testBucket/Test/a.txt, Type - File
+	if obj[0].Name() != FileNameInDirectoryTestBucket || obj[0].IsDir() != false {
+		t.Errorf("Listed object from bucket directory is incorrect.")
```
Comment (sethiay) [inline tools/integration_tests/readonly/list_objects_test.go:76]: nit: Object Listed for file in bucket directory is incorrect.

File: tools/integration_tests/readonly/list_objects_test.go [hunk 8]
```diff
@@ -60,15 +65,20 @@ func TestListObjectsInBucketSubDirectory(t *testing.T) {
...
 	}
 
-	if obj[1].Name() != SubDirectoryNameInTestBucket && obj[1].IsDir() != true {
-		t.Errorf("Listed object is incorrect.")
+	// Comparing second object name and type.
+	// Name - testBucket/b, Type - Dir
+	if obj[1].Name() != SubDirectoryNameInTestBucket || obj[1].IsDir() != true {
+		t.Errorf("Listed object from bucket directory is incorrect.")
```
Comment (sethiay) [inline tools/integration_tests/readonly/list_objects_test.go:82]: nit: Object Listed for bucket sub directory is incorrect.

File: tools/integration_tests/readonly/list_objects_test.go [hunk 9]
```diff
@@ -35,20 +35,25 @@ func TestListObjectsInBucket(t *testing.T) {
...
+	// Name - testBucket/Test, Type - Dir
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
 		t.Errorf("Listed object is incorrect.")
 	}
 
-	if obj[1].Name() != FileNameInTestBucket && obj[1].IsDir() != false {
+	// Comparing second object name and type
+	// Name - testBucket/Test1.txt, Type - File
+	if obj[1].Name() != FileNameInTestBucket || obj[1].IsDir() != false {
 		t.Errorf("Listed object is incorrect.")
```
Comment (sethiay) [inline tools/integration_tests/readonly/list_objects_test.go:52]: nit: Object Listed for file in bucket is incorrect.

File: tools/integration_tests/readonly/list_objects_test.go [hunk 10]
```diff
@@ -35,20 +35,25 @@ func TestListObjectsInBucket(t *testing.T) {
...
+	// Comparing number of objects in the testBucket - 2
 	if len(obj) != NumberOfObjectsInTestBucket {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != DirectoryNameInTestBucket && obj[0].IsDir() != true {
+	// Comparing first object name and type
+	// Name - testBucket/Test, Type - Dir
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
 		t.Errorf("Listed object is incorrect.")
```
Comment (sethiay) [inline tools/integration_tests/readonly/list_objects_test.go:46]: nit: Object Listed for bucket directory is incorrect.

**Review by sethiay — APPROVED** [review]

File: tools/integration_tests/readonly/list_objects_test.go (hunk 7 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: fixed

File: tools/integration_tests/readonly/list_objects_test.go (hunk 9 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: fixed

File: tools/integration_tests/readonly/list_objects_test.go (hunk 10 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: fixed

File: tools/integration_tests/readonly/list_objects_test.go (hunk 8 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: fixed

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**